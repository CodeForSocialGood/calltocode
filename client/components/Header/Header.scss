@import '../../scss/partials/partials';

.headerRoot {
  width: 100%;

  &:after {
    content: '';
    position: absolute;
    width: 100%;
    top: 64px;
    left: 0;
    height: 2px;
    background: #F9F9F9;
  }
}

.flex {
  flex: 1;
}

.button {
  padding: 10px;
  margin-left: 10px;
  color: dimgray;

  &.active, &:active {
    @extend %buttonActive;
  }

  &:hover {
    @extend %hover;
  }
}
