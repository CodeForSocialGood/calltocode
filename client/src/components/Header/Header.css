.header {
  text-align: right;
  padding: 0 60px;
  margin-top: 20px;
}

.button {
  padding: 10px;
  border: 0;
  background-color: royalblue;
  color: white;
  font-size: 100%;
  border-radius: 5px;
}

.button:first-child {
  margin-right: 10px;
}
