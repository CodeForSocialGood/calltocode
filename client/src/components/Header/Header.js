import styles from './Header.css'
import React from 'react'

import { Link } from 'react-router-dom'

function Header () {
  return (
    <header className={styles.header}>
      <Link to='/' className={styles.button}>SIGN UP</Link>
      <Link to='/login' className={styles.button}>LOGIN</Link>
    </header>
  )
}

export default Header
