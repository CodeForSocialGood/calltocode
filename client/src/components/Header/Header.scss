@import '../../scss/partials/partials';

.header {
  padding: 0 60px;
  margin-top: 60px;
  display: grid;
  grid-template-columns: repeat(2, 1fr);
}

.button {
  @extend %buttonSub;
  padding: 10px;
  margin-left: 10px;

  &:hover {
    @extend %hover;
  }
}

.buttons {
  text-align: right;
}

.logo {
  height: 42px;
}
