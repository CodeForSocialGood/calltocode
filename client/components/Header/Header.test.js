/* eslint-env jest */
import React from 'react'
import {shallow} from 'enzyme'

import {Header} from './Header'
let mockProps

beforeEach(() => {
  mockProps = {
    authenticated: null,
    currentPage: '',
    logout: () => {},
    user: {
      usertype: null
    }
  }
})

describe('<Header />', () => {
  describe('authenticated is truthy', () => {
    beforeEach(() => { mockProps.authenticated = true })

    describe('currentPage includes the string "profile"', () => {
      beforeEach(() => { mockProps.currentPage += 'profile' })

      describe('user\'s usertype is "contact"', () => {
        beforeEach(() => { mockProps.user.usertype = 'contact' })

        describe('currentPage includes the string "create-project"', () => {
          beforeEach(() => { mockProps.currentPage += 'create-project' })

          test('component renders correctly', () => {
            const component = shallow(<Header {...mockProps} />)
            expect(component).toMatchSnapshot()
          })
        })

        describe('currentPage does not include the string "create-project"', () => {
          test('component renders correctly', () => {
            const component = shallow(<Header {...mockProps} />)
            expect(component).toMatchSnapshot()
          })
        })
      })

      describe('user\'s usertype is not "contact"', () => {
        beforeEach(() => { mockProps.user.usertype = 'not_contact' })

        test('component renders correctly', () => {
          const component = shallow(<Header {...mockProps} />)
          expect(component).toMatchSnapshot()
        })
      })
    })
  })

  describe('authenticated is falsey', () => {
    beforeEach(() => { mockProps.authenticated = false })

    describe('currentPage includes the string "signup"', () => {
      beforeEach(() => { mockProps.currentPage += 'signup' })

      describe('currentPage includes the string "login"', () => {
        beforeEach(() => { mockProps.currentPage += 'login' })

        test('component renders correctly', () => {
          const component = shallow(<Header {...mockProps} />)
          expect(component).toMatchSnapshot()
        })
      })

      describe('currentPage does not include the string "login"', () => {
        test('component renders correctly', () => {
          const component = shallow(<Header {...mockProps} />)
          expect(component).toMatchSnapshot()
        })
      })
    })

    describe('currentPage does not include the string "signup"', () => {
      describe('currentPage includes the string "login"', () => {
        beforeEach(() => { mockProps.currentPage += 'login' })

        test('component renders correctly', () => {
          const component = shallow(<Header {...mockProps} />)
          expect(component).toMatchSnapshot()
        })
      })

      describe('currentPage does not include the string "login"', () => {
        test('component renders correctly', () => {
          const component = shallow(<Header {...mockProps} />)
          expect(component).toMatchSnapshot()
        })
      })
    })
  })
})
