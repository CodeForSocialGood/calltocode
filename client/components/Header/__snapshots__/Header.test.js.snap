// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> authenticated is falsey currentPage does not include the string "signup" currentPage does not include the string "login" component renders correctly 1`] = `
<div
  className="headerRoot"
>
  <WithStyles(AppBar)
    color="inherit"
    elevation={0}
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Typography)
        className="flex"
        color="inherit"
        type="title"
      >
        <Link
          className="button"
          replace={false}
          to="/"
        >
          calltocode
        </Link>
      </WithStyles(Typography)>
      <Link
        className="button"
        key="signup"
        replace={false}
        to="/signup"
      >
        SIGN UP
      </Link>
      <Link
        className="button"
        key="login"
        replace={false}
        to="/login"
      >
        LOG IN
      </Link>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;

exports[`<Header /> authenticated is falsey currentPage does not include the string "signup" currentPage includes the string "login" component renders correctly 1`] = `
<div
  className="headerRoot"
>
  <WithStyles(AppBar)
    color="inherit"
    elevation={0}
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Typography)
        className="flex"
        color="inherit"
        type="title"
      >
        <Link
          className="button"
          replace={false}
          to="/"
        >
          calltocode
        </Link>
      </WithStyles(Typography)>
      <Link
        className="button"
        key="signup"
        replace={false}
        to="/signup"
      >
        SIGN UP
      </Link>
      <Link
        className="button active"
        key="login"
        replace={false}
        to="/login"
      >
        LOG IN
      </Link>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;

exports[`<Header /> authenticated is falsey currentPage includes the string "signup" currentPage does not include the string "login" component renders correctly 1`] = `
<div
  className="headerRoot"
>
  <WithStyles(AppBar)
    color="inherit"
    elevation={0}
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Typography)
        className="flex"
        color="inherit"
        type="title"
      >
        <Link
          className="button"
          replace={false}
          to="/"
        >
          calltocode
        </Link>
      </WithStyles(Typography)>
      <Link
        className="button active"
        key="signup"
        replace={false}
        to="/signup"
      >
        SIGN UP
      </Link>
      <Link
        className="button"
        key="login"
        replace={false}
        to="/login"
      >
        LOG IN
      </Link>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;

exports[`<Header /> authenticated is falsey currentPage includes the string "signup" currentPage includes the string "login" component renders correctly 1`] = `
<div
  className="headerRoot"
>
  <WithStyles(AppBar)
    color="inherit"
    elevation={0}
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Typography)
        className="flex"
        color="inherit"
        type="title"
      >
        <Link
          className="button"
          replace={false}
          to="/"
        >
          calltocode
        </Link>
      </WithStyles(Typography)>
      <Link
        className="button active"
        key="signup"
        replace={false}
        to="/signup"
      >
        SIGN UP
      </Link>
      <Link
        className="button active"
        key="login"
        replace={false}
        to="/login"
      >
        LOG IN
      </Link>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;

exports[`<Header /> authenticated is truthy currentPage includes the string "profile" user's usertype is "contact" currentPage does not include the string "create-project" component renders correctly 1`] = `
<div
  className="headerRoot"
>
  <WithStyles(AppBar)
    color="inherit"
    elevation={0}
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Typography)
        className="flex"
        color="inherit"
        type="title"
      >
        <Link
          className="button"
          replace={false}
          to="/"
        >
          calltocode
        </Link>
      </WithStyles(Typography)>
      <Link
        className="button"
        key="create-project"
        replace={false}
        to="/create-project"
      >
        CREATE PROJECT
      </Link>
      <Link
        className="button"
        key="logout"
        onClick={[Function]}
        replace={false}
        to="/"
      >
        LOG OUT
      </Link>
      <Link
        className="button active"
        key="profile"
        replace={false}
        to="/profile"
      >
        PROFILE
      </Link>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;

exports[`<Header /> authenticated is truthy currentPage includes the string "profile" user's usertype is "contact" currentPage includes the string "create-project" component renders correctly 1`] = `
<div
  className="headerRoot"
>
  <WithStyles(AppBar)
    color="inherit"
    elevation={0}
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Typography)
        className="flex"
        color="inherit"
        type="title"
      >
        <Link
          className="button"
          replace={false}
          to="/"
        >
          calltocode
        </Link>
      </WithStyles(Typography)>
      <Link
        className="button active"
        key="create-project"
        replace={false}
        to="/create-project"
      >
        CREATE PROJECT
      </Link>
      <Link
        className="button"
        key="logout"
        onClick={[Function]}
        replace={false}
        to="/"
      >
        LOG OUT
      </Link>
      <Link
        className="button active"
        key="profile"
        replace={false}
        to="/profile"
      >
        PROFILE
      </Link>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;

exports[`<Header /> authenticated is truthy currentPage includes the string "profile" user's usertype is not "contact" component renders correctly 1`] = `
<div
  className="headerRoot"
>
  <WithStyles(AppBar)
    color="inherit"
    elevation={0}
    position="static"
  >
    <WithStyles(Toolbar)>
      <WithStyles(Typography)
        className="flex"
        color="inherit"
        type="title"
      >
        <Link
          className="button"
          replace={false}
          to="/"
        >
          calltocode
        </Link>
      </WithStyles(Typography)>
      <Link
        className="button"
        key="logout"
        onClick={[Function]}
        replace={false}
        to="/"
      >
        LOG OUT
      </Link>
      <Link
        className="button active"
        key="profile"
        replace={false}
        to="/profile"
      >
        PROFILE
      </Link>
    </WithStyles(Toolbar)>
  </WithStyles(AppBar)>
</div>
`;
